.github/workflows/publish-olm-bundle.yaml

```
@@ -0,0 +1,162 @@
+          if [[ "$title" =~ ^'OLM test submit v' ]]; then
+            USER_REPO="${{ github.event.issue.user.login }}/community-operators"
+            echo "repo=$USER_REPO" >> $GITHUB_OUTPUT
+            echo "fork=$BOT_REPO" >> $GITHUB_OUTPUT
+            echo "tag=${title#OLM test submit }" >> $GITHUB_OUTPUT
+            exit 0
+          fi
+
+          # Otherwise skip.
+          echo "skip=true" >> $GITHUB_OUTPUT
+
+      - name: Fail if BOT_PAT is unset
+        if: ${{ steps.check.outputs.skip != 'true' && env.BOT_PAT == '' }}
+        run: |
+          echo "Bot personal access token (BOT_PAT) not set."
+          exit 1
+
+      - name: Fail if BOT_GPG_PRIVATE_KEY is unset
+        if: ${{ steps.check.outputs.skip != 'true' && env.BOT_GPG_PRIVATE_KEY == '' }}
+        run: |
+          echo "Bot personal access token (BOT_GPG_PRIVATE_KEY) not set."
+          exit 1
+
+      - name: Fail is BOT_GPG_PASSPHRASE is unset
+        if: ${{ steps.check.outputs.skip != 'true' && env.BOT_GPG_PASSPHRASE == '' }}
+        run: |
+          echo "Bot personal access token (BOT_GPG_PASSPHRASE) not set."
+          exit 1
+
+      - name: Check PR target repository exists
+        if: ${{ steps.check.outputs.skip != 'true' }}
+        run: |
+         # Check that the PR target repository exists
+         repo="${{ steps.check.outputs.repo }}"
+         repo_status=$(curl -s -o /dev/null -I -w "%{http_code}" "https://github.com/$repo")
+         echo "repo: $repo, repo HTTP access status: $repo_status"
+         if [ "$repo_status" != "200" ]; then
+             echo "Target repo ($repo) does not exist (HTTP status $repo_status)."
+             exit 1
+         fi
+
+  createPullRequest:
+    name: Publish new OperatorHub release
+    runs-on: ubuntu-22.04
+    needs: trigger
+    if: ${{ needs.trigger.outputs.skip != 'true' }}
+    env:
+      REPO: ${{ needs.trigger.outputs.repo }}
+      FORK: ${{ needs.trigger.outputs.fork }}
+      TAG: ${{ needs.trigger.outputs.tag }}
+    steps:
+      - name: Show REPO, FORK, and TAG being used
+        run: |
+          echo "REPO: ${{ env.REPO }}"
+          echo "FORK: ${{ env.FORK }}"
+          echo "TAG: ${{ env.TAG }}"
+
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Build the bundle
+        run: |
+          version="${{ env.TAG }}"
+          version="${version#v}"
+          pushd deployment/operator && VERSION=${version} make bundle && popd
+
+      - name: Checkout upstream community-operators repo
+        uses: actions/checkout@v4
+        with:
+          repository: ${{ env.REPO }}
+          path: community-operators
+          ref: main
+          token: ${{ secrets.BOT_PAT }}
+
+      - name: Import GPG key
+        uses: crazy-max/ghaction-import-gpg@v6
+        id: import-gpg
+        with:
+          gpg_private_key: ${{ secrets.BOT_GPG_PRIVATE_KEY }}
+          passphrase: ${{ secrets.BOT_GPG_PASSPHRASE }}
```

```suggestion
          passphrase: ${{ secrets.BOT_GPG_PASSPHRASE }}
          fingerprint: "15CF4622AD8FE3794F15EED3E3063FF042266F81"
```
@kad has asked if it would be possible to use the subKey. It is possible via the fingerprint field and this an actual fingerprint for the same secret stored on GitHub.

---

hardcoding that value here doesn't make sense. better to have it as a separate secret configured in repo (if really needed)

---

Well I am just telling that it could be done this way since you asked.

---

I think @klihub  already did that in the repo settings, and gpg_private_key is pointing to signing subkey.

---

Have you tested that you can use the subkey via `gpg_private_key` instead of using `fingerprint` ?

---

You could have asked me for the subkey. Seems I just wasted some time. But as long as things work as you expected i'm fine with that.